### PR TITLE
Feature - Combo - Add support for combo input-group + clear & toggle button templating

### LINKED ISSUE #3613: Feature - Combo - Add support for input group customization, add templating of Clear and Toggle buttons
### Description  
The `IgxComboComponent` does not currently support usage of custom `igx-prefix`, `igx-suffix`, `igx-hint` elements and the use of `IgxLabel` directive to enhance the combo `input` element.

Currently the combo `clear` and `toggle` button content cannot be templated

 * igniteui-angular version: 7.2.x
 * browser: N/A

### Expected result  
The `IgxComboComponent` should support use of custom `igx-prefix`, etc. in order to enhance the combo input-group 

Allow templating of the combo `clear` and `toggle` button content (using directives, similar to `igxComboAddItem`)

### PR DESCRIPTION
**Pull request by ViktorSlavov**
Closes #3613 

### Additional information (check all that apply):
 - [x] New functionality

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes `feature/README.MD` updates for the feature docs
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality

### CODE REVIEW

File: projects/igniteui-angular/src/lib/combo/combo.component.html [hunk 1]
```diff
@@ -32,7 +32,7 @@
         </ng-container>
         <igx-icon *ngIf="!clearIconTemplate" fontSet="material">clear</igx-icon>
     </igx-suffix>
-    <igx-suffix igxButton="icon" class="igx-combo__clear-button" igxRipple>
+    <igx-suffix igxButton="icon" class="clearButton" igxRipple>
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/combo/combo.component.html]: Why is this change needed?

File: projects/igniteui-angular/src/lib/combo/combo.component.html [hunk 2]
```diff
@@ -15,32 +15,47 @@
...
+        <ng-container *ngIf="clearIconTemplate">
+            <ng-container *ngTemplateOutlet="clearIconTemplate; context: {$implicit: this}"></ng-container>
+        </ng-container>
+        <igx-icon *ngIf="!clearIconTemplate" fontSet="material">clear</igx-icon>
     </igx-suffix>
-    <igx-suffix igxButton="icon" class="igx-combo__clear-button" igxRipple>
-        <igx-icon fontSet="material">{{ dropdown.collapsed ? 'arrow_drop_down' : 'arrow_drop_up'}}</igx-icon>
+    <igx-suffix igxButton="icon" class="igx-combo__toggle-button" igxRipple>
+        <ng-container *ngIf="toggleIconTemplate">
+            <ng-container *ngTemplateOutlet="toggleIconTemplate; context: {$implicit: this}"></ng-container>
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/combo/combo.component.html]: I think @rkaraivanov will agree, when I say templates shouldn't expose more than it's required.
You only need the collapsed state for ease of use here, so you should probably only provide `context: {$implicit: this.collapsed}`.

File: projects/igniteui-angular/src/lib/combo/combo.component.html [hunk 3]
```diff
@@ -15,32 +15,47 @@
...
-        [disabled]="disabled"/>
-    <igx-suffix *ngIf="value.length" class="clearButton" aria-label="Clear Selection" igxRipple (click)="handleClearItems($event)">
-        <igx-icon fontSet="material">clear</igx-icon>
+        [disabled]="disabled" />
+    <ng-container ngProjectAs="igx-suffix">
+        <ng-content select="igx-suffix"></ng-content>
+    </ng-container>
+    <igx-suffix *ngIf="value.length" aria-label="Clear Selection" class="igx-combo__clear-button" igxRipple (click)="handleClearItems($event)">
+        <ng-container *ngIf="clearIconTemplate">
+            <ng-container *ngTemplateOutlet="clearIconTemplate; context: {$implicit: this}"></ng-container>
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/combo/combo.component.html]: Similar logic applies here and this probably doesn't need any additional context at all.
Same with others like the header, add item, etc.

File: projects/igniteui-angular/src/lib/combo/combo.component.html (hunk 2 above)
Comment (rkaraivanov) [reply projects/igniteui-angular/src/lib/combo/combo.component.html]: 👍

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,25 @@ All notable changes for each version of this project will be documented in this
...
+            <label igxLabel>Genres</label>
+            <igx-prefix><igx-icon>music_note</igx-icon></igx-prefix>
+        </igx-combo>
+     ```
+    - **Feature** the default combo 'clear' and 'toggle' icons can now be templated. Two new directives are added (with selector `[igxComboClearIcon]` and `[igxComboToggleIcon]`). Passing an `ng-template` with one of the directives will overwrite the default conent of the respective icon. Functionality will remain unaffected. Expample:
+    ```html
+        <!-- customize combo input --->
+        <igx-combo #myCombo [data]="myGenres">
+            ...
+            <ng-template igxComboToggleIcon let-item>
```
Comment (damyanpetev) [inline CHANGELOG.md]:
```suggestion
            <ng-template igxComboToggleIcon let-collapsed>
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,25 @@ All notable changes for each version of this project will be documented in this
...
+                <igx-icon>{{ item.collapsed ? 'remove_circle' : 'remove_circle_outline'}}</igx-icon>
```
Comment (damyanpetev) [inline CHANGELOG.md]:
```suggestion
                <igx-icon>{{ collapsed ? 'remove_circle' : 'remove_circle_outline'}}</igx-icon>
```
